javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/AgentListener.java

```diff
@@ -25,4 +30,21 @@ public interface AgentListener extends Ordered {
    * on an {@link Instrumentation}.
    */
   void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk);
+
+  /** Resolve {@link ConfigProperties} from the {@code autoConfiguredOpenTelemetrySdk}. */
+  static ConfigProperties resolveConfigProperties(
+      AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    ConfigProperties sdkConfigProperties =
+        AutoConfigureUtil.getConfig(autoConfiguredOpenTelemetrySdk);
+    if (sdkConfigProperties != null) {
+      return sdkConfigProperties;
+    }
+    StructuredConfigProperties structuredConfigProperties =
+        AutoConfigureUtil.getStructuredConfig(autoConfiguredOpenTelemetrySdk);
+    if (structuredConfigProperties != null) {
+      return new StructuredConfigPropertiesBridge(structuredConfigProperties);
+    }
+    // Should never happen
```

Comment:
Perhaps it would be worth to print something on the standard output or fail in this case?

Reply:
if the code is in an invalid state, I would prefer to be more explicit in the code (exception or something like that), and for users if unfortunately, for some reasons, it happens.

Reply:
I'm happy to throw as well. For context, the guarantee that `AutoConfiguredOpenTelemetrySdk` has either `ConfigProperties` or `StructuredConfigProperties` is ensured by the autoconfigure module. Reaching this point means there is a bug in the autoconfigure module.